src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -360,13 +360,65 @@ private BoundStatement BindThrow(ThrowStatementSyntax node, DiagnosticBag diagno
             return new BoundThrowStatement(node, boundExpr, hasErrors);
         }
 
-        private static BoundStatement BindEmpty(EmptyStatementSyntax node)
+        private static bool IsEmbeddedStatementParent(SyntaxNode node)
         {
+            if (node != null)
+            {
+                switch (node.Kind())
+                {
+                    case SyntaxKind.FixedStatement:
+                    case SyntaxKind.DoStatement:
+                    case SyntaxKind.IfStatement:
+                    case SyntaxKind.ElseClause:
+                    case SyntaxKind.LockStatement:
+                    case SyntaxKind.UsingStatement:
+                    case SyntaxKind.ForStatement:
+                    case SyntaxKind.ForEachStatement:
+                    case SyntaxKind.WhileStatement:
+                        return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static BoundStatement BindEmpty(EmptyStatementSyntax node, DiagnosticBag diagnostics)
+        {
+            if (IsEmbeddedStatementParent(node.Parent))
```

Comment:
>if (IsEmbeddedStatementParent(node.Parent)) [](start = 12, length = 43)

It feels like it would be better to intercept this statement in BindPossibleEmbeddedStatement function instead.